pkg/cloud/services/iamauth/reconcile.go

```diff
@@ -124,7 +125,9 @@ func (s *Service) getRolesForMachineDeployments(ctx context.Context, allRoles ma
 	}
 	err := s.client.List(ctx, deploymentList, selectors...)
 	if err != nil {
-		return fmt.Errorf("failed to list machine deployments for cluster %s/%s: %w", s.scope.Namespace(), s.scope.Name(), err)
+		if !containsTimeoutSyncError(err) {
```

Comment:
I'm not convinced that this is the solution to the issue. 

It would be good to understand by the informer isn't synced. Do we need to introduce changes to the `Cache` settings for the manager? Would indexers help?